pkg/resource/deploy/import.go

```diff
@@ -342,12 +341,8 @@ func (i *importer) registerProviders(ctx context.Context) (map[resource.URN]stri
 	// Update the URN to reference map.
 	for _, s := range steps {
 		res := s.Res()
-		id := res.ID
-		if i.preview {
-			id = providers.UnknownID
-		}
```

Comment:
Is this condition being lost of no importance?

Reply:
Nothing ever set preview, so it was always false, so this was always just res.ID